sklearn/naive_bayes.py

```diff
@@ -200,14 +207,28 @@ def _update_mean_variance(n_past, mu, var, X):
             Updated variance for each Gaussian over the combined set.
         """
         if n_past == 0:
-            return np.mean(X, axis=0), np.var(X, axis=0)
+            if sample_weight is not None:
+                sum_sw = float(sample_weight.sum())
+                mu = np.average(X, axis=0, weights=sample_weight / sum_sw)
+                var = np.average((X - mu) ** 2, axis=0,
+                                  weights=sample_weight / sum_sw)
+                return mu, var
+            else:
+                return np.mean(X, axis=0), np.var(X, axis=0)
         elif X.shape[0] == 0:
             return mu, var
 
         old_ssd = var * n_past
-        n_new = X.shape[0]
-        new_ssd = n_new * np.var(X, axis=0)
-        new_sum = np.sum(X, axis=0)
+        if sample_weight is not None:
+            n_new = float(sample_weight.sum())
```

Comment:
can you refactor this to not repeat it? Or do you think that will not make it clearer?